### PR TITLE
Import GitHub issues

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/commands/github/issue/create_or_update.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          i.opened_at = attributes[:opened_at]
+          i.opened_by_username = attributes[:opened_by_username]
+        end
+
+        issue.update!(
+          number: attributes[:number],
+          title: attributes[:title],
+          status: status,
+          repo: attributes[:repo],
+          opened_at: attributes[:opened_at],
```
Comment (ErikSchierboom) [inline app/commands/github/issue/create_or_update.rb:23]: I took the liberty of also adding this field (it wasn't in our "spec")

File: app/commands/github/issue/destroy.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Github
+  class Issue
+    class Destroy
```
Comment (ErikSchierboom) [inline app/commands/github/issue/destroy.rb:3]: We use this only when we get the `deleted` event from the webhook. We could also treat deleted issues as closed 🤷

File: app/controllers/webhooks/issue_updates_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# This controller listens for Github pull request webhook events,
+# for which the "Issues" event type must be enabled in GitHub.
+# See https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#issues
+module Webhooks
+  class IssueUpdatesController < BaseController
```
Comment (ErikSchierboom) [inline app/controllers/webhooks/issue_updates_controller.rb:5]: To get this working, remember to set a global webhook that listens for the "Issues" event type and points to `/webhooks/issue_updates`

File: app/models/github/issue.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Github::Issue < ApplicationRecord
+  enum status: { open: 0, closed: 1 }, _prefix: true
+
+  has_many :labels,
+    dependent: :destroy,
+    inverse_of: :issue,
+    class_name: "Github::IssueLabel",
+    foreign_key: "github_issue_id"
+
+  scope :open, -> { where(status: :open) }
```
Comment (ErikSchierboom) [inline app/models/github/issue.rb:10]: I assumed that we usually only want to deal with open issues, hence the scope

File: config/sidekiq-schedule.yml [hunk 5]
```diff
@@ -7,3 +7,8 @@ fetch_and_sync_all_pull_requests_reputation:
   cron: "0 1 * * 1"
   class: "FetchAndSyncAllPullRequestsReputationJob"
   queue: dribble
+
+sync_issues:
+  cron: "0 2 * * 1"
```
Comment (ErikSchierboom) [inline config/sidekiq-schedule.yml]: I've started this job one hour after the other job, which is enough time for it to finish I think.

Comment (iHiD) [reply config/sidekiq-schedule.yml]: is it worth doing on a different day so that we don't overload our token?

Comment (ErikSchierboom) [reply config/sidekiq-schedule.yml]: Good idea

Comment (ErikSchierboom) [inline config/sidekiq-schedule.yml]:
```suggestion
  cron: "0 1 * * 2"
```

Comment (ErikSchierboom) [reply config/sidekiq-schedule.yml]: It now runs a day later. Theoretically it shouldn't really matter as tokens reset each hour, but it is better to spread these jobs around regardless.

**Pull request merged**